### PR TITLE
Fix flag for library dataset option in data selector

### PR DESCRIPTION
**Pull request by guerler**

### CODE REVIEW

File: client/galaxy/scripts/mvc/ui/ui-select-content.js [hunk 1]
```diff
@@ -7,16 +7,17 @@ import Ui from "mvc/ui/ui-misc";
 import Select from "mvc/ui/ui-select-default";
 import { getGalaxyInstance } from "app";
 
-/** Batch mode variations */
-var Batch = { DISABLED: "disabled", ENABLED: "enabled", LINKED: "linked" };
+/** Batch mode letiations */
+let Batch = { DISABLED: "disabled", ENABLED: "enabled", LINKED: "linked" };
```
Comment (dannon) [inline client/galaxy/scripts/mvc/ui/ui-select-content.js]: We should prefer `const` for situations like this where there is no rebinding of the variables.

**Comment by martenson** [comment]
Why are we restricting anons here?

File: client/galaxy/scripts/mvc/ui/ui-select-content.js (hunk 1 above)
Comment (guerler) [reply client/galaxy/scripts/mvc/ui/ui-select-content.js]: Thanks @dannon.

**Comment by guerler** [comment]
@martenson, as of right now, `http://127.0.0.1:8080/api/libraries/f2db41e1fa331b3e/contents` is not exposed for anon-users. It triggers a forbidden exception.

**Comment by martenson** [comment]
We should fix that instead of restricting this feature. I will have a look
tomorrow.

File: client/galaxy/scripts/mvc/ui/ui-select-content.js (hunk 1 above)
Comment (dannon) [reply client/galaxy/scripts/mvc/ui/ui-select-content.js]: @guerler Sure, no worries.  I just opened https://github.com/galaxyproject/galaxy/pull/7860 that'll fix the existing occurrences of this and will lint it moving forward, too.

**Comment by guerler** [comment]
@martenson, I removed that particular setting from this PR. The remaining PR is still relevant, it fixes the overall handling of that option. Thank you for the review.

**Comment by martenson** [comment]
~~You might need to rebase because of https://github.com/galaxyproject/galaxy/pull/7860~~

You actually need to switch everything to `const` :)

**Pull request merged**